src/fstests/runner.py

```diff
@@ -67,60 +67,42 @@ def _run_single_job(self, tarball_node, device):
         finally:
             return True
 
-    def check_kvm_xfstest_env(self):
-        try:
-            result = subprocess.run(
-                "kvm-xfstests --help", check=True, shell=True)
-            if result.returncode == 0:
-                print(f"KVM-xftests found.")
-                return True
-            else:
-                return False
-        except Exception as e:
-            print(f"Raised exception: {e}")
+    def run(self, args):
+        if args.node_id:
+            tarball_node = self._db.get_node(args.node_id)
+        else:
+            tarball_node = None
+        if tarball_node is None:
+            print("Node not found")
             return False
-
-    def run(self):
-        sub_id = self._db.subscribe_node_channel({
-            'op': 'created',
-            'name': 'tarball',
-        })
-        print('Listening for tarballs')
-        print('Press Ctrl-C to stop.')
-        try:
-            while True:
-                tarball_node = self._db.receive_node(sub_id)
-                print(f"Node tarball with id: {tarball_node['_id']}\
-                   from revision: {tarball_node['revision']['commit'][:12]}")
-                device = self._device_configs['shell']
-                self._run_single_job(tarball_node, device)
-        except KeyboardInterrupt as e:
-            print('Stopping.')
-        except Exception as e:
-            print('Error', e)
-        finally:
-            self._db.unsubscribe(sub_id)
-            return True
-
-
-class cmd_check(Command):
-    help = 'Check KVM-xfstests environment'
-    args = [Args.db_config]
-    opt_args = [Args.verbose]
-
-    def __call__(self, configs, args):
-        return FstestsRunner(configs, args).check_kvm_xfstest_env()
+        print(f"Node tarball with id: {tarball_node['_id']}\
+        from revision: {tarball_node['revision']['commit'][:12]}")
+        device = self._device_configs['shell']
+        return self._run_single_job(tarball_node, device)
 
 
 class cmd_run(Command):
     help = 'KVM fstests runner'
     args = [
         Args.db_config, Args.output,
     ]
-    opt_args = [Args.verbose]
+    opt_args = [
+        Args.verbose,
+        {
+            'name': '--node-id',
+            'help': "id of the checkout node rather than pub/sub",
+        },
+    ]
 
     def __call__(self, configs, args):
-        return FstestsRunner(configs, args).run()
+        if not args.node_id:
+            print("--node-id is required",
+                  file=sys.stderr)
+        try:
+            return FstestsRunner(configs, args).run(args)
+        except Exception as e:
+            print(f"Error: {e}", file=sys.stderr)
```

Comment:
All this can be dropped.  The `--node-id` argument is optional, if you wanted it to be required it would be listed in `args` rather than `opt_args` in the class.  I think it's fine to have it optional, so that when it's provided you use it and when it's not provided then you listen to the pub/sub interface to automatically get events about new tarball nodes.